web/pages/dashboard/[...slugs]/index.tsx

```diff
@@ -0,0 +1,108 @@
+import { Page } from 'web/components/page'
+
+import { fromPropz, usePropz } from 'web/hooks/use-propz'
+import { dashboardPath, getDashboardBySlug } from 'web/lib/firebase/dashboards'
+import { Dashboard } from 'common/dashboard'
+import { Title } from 'web/components/title'
+import { Spacer } from 'web/components/layout/spacer'
+import { Content } from 'web/components/editor'
+import { UserLink } from 'web/components/user-page'
+import { getUser, User } from 'web/lib/firebase/users'
+import { ShareIcon } from '@heroicons/react/solid'
+import clsx from 'clsx'
+import { Button } from 'web/components/button'
+import { useState } from 'react'
+import { ShareDashboardModal } from 'web/components/share-dashboard-modal'
+import { Row } from 'web/components/layout/row'
+import { Col } from 'web/components/layout/col'
+import { ENV_CONFIG } from 'common/envs/constants'
+import Custom404 from 'web/pages/404'
+
+export const getStaticProps = fromPropz(getStaticPropz)
+export async function getStaticPropz(props: { params: { slugs: string[] } }) {
+  const { slugs } = props.params
+
+  const dashboard = await getDashboardBySlug(slugs[0])
+  const creatorPromise = dashboard ? getUser(dashboard.creatorId) : null
+  const creator = await creatorPromise
+
+  return {
+    props: {
+      dashboard: dashboard,
+      creator: creator,
+    },
+
+    revalidate: 60, // regenerate after a minute
+  }
+}
+
+export async function getStaticPaths() {
+  return { paths: [], fallback: 'blocking' }
+}
+
+export default function DashboardPage(props: {
+  dashboard: Dashboard
+  creator: User
+}) {
+  props = usePropz(props, getStaticPropz) ?? {
+    dashboard: null,
```

Comment:
Might want to start as `dashboard: undefined, creator: undefined`.

We often use the pattern of `undefined` meaning that it hasn't loaded yet, while `null` means that it's loaded and was not found.